lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
```

Comment:
If this comment is reflective of the code below, then it's not what we talked about.
We want to apply the bulk update to all of the documents, regardless of what they've done; we're just going to remove `_rev` from the `docs` entries so that we don't force-overwrite items whose `_rev` values might be newer/different.

By doing that, we should be covered for these scenarios:
`_rev` is missing from one or more objects in `data`: They'll fail the rev-check in Cloudant, and will be aggregated in the list of failures returned to us in the results payload.
`_rev` is on all of the entries in `data`: Cloudant will still aggregate any conflicting documents together, and then we'll be able to return an error that includes the items that failed.

The downside to this is that we have to return an error code even though some of the bulk update succeeded (which is in contrast with Cloudant's return of a 201 or 202 depending on *how* successful your bulk update was). 
Our hands are tied in this regard due to the LoopBack API's expectation that updateAll returns a record *count*, and not a full list of items updated.

**TL;DR** - Remove this comment, make sure the code isn't doing what this comment says it's doing.